cpp/include/cudf/reduction/detail/reduction.cuh

```diff
@@ -65,8 +66,10 @@ std::unique_ptr<scalar> reduce(InputIterator d_in,
 {
   auto const binary_op     = cudf::detail::cast_functor<OutputType>(op.get_binary_op());
   auto const initial_value = init.value_or(op.template get_identity<OutputType>());
+  auto pinned_initial = cudf::detail::make_pinned_vector_async<OutputType>(1, stream);
+  pinned_initial[0] = initial_value;
   using ScalarType         = cudf::scalar_type_t<OutputType>;
-  auto result              = std::make_unique<ScalarType>(initial_value, true, stream, mr);
+  auto result = std::make_unique<ScalarType>(pinned_initial[0], true, stream, mr);
```

Comment:
> I forgot most of the context here :(
are we passing the value by reference here?

(Sorry, I couldn’t respond to your review here because I did a new commit affecting those lines.)

Let me bring back some context from our Slack chat. The goal is for `ScalarType` and `cub::DeviceReduce::Reduce` to copy the `initial_value` from host-pinned memory.

Back around August 19th in Slack, we discussed:
- placing the `initial_value` in a pinned host vector of size 1
- and then assigning the value to the first element [0].